### ISSUE TITLE
[NPEB01352] Deadly Premonition: Director's Cut - Segfault with OpenGL on Mesa

### ISSUE THREAD

**Issue by valters-tomsons** [RPCS3/rpcs3#15590]
### Quick summary

Trying to run the game with OpenGL renderer causes a segfault shortly after loading a save game. Does not happen on Vulkan.

### Details

I've been seeing this issue since December 2023, on various RPCS3/Mesa/kernel versions released since then. 

1. Download and run the latest AppImage
1. Configure game to use OpenGL & boot it
1. When you reach main menu, load game (or start a new one)
1. Cutscene(s) will play, feel free to skip them
1. Crash occurs, see log

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/15312057/RPCS3.log)

### System configuration

CPU: Ryzen 7 5800X
GPU: Radeon RX 6700 / Mesa `24.0.6-2`
OS: Arch Linux `6.8.9`
DE: KDE Plasma 6 (Wayland)

**Comment by kd-11** [comment]
Cannot reproduce on mesa-git (24.2.0-devel)

**Comment by valters-tomsons** [comment]
I was wrong earlier, this does not seem to happen when starting a new game. Will try to reproduce from a new game & upload a save game.

**Comment by valters-tomsons** [comment]
I lost my previous save game, which used to crash instantly. It's still reproducible in the Prologue, but it takes up to 10 minutes for the crash to happen.

I've had two crashes with the save below, one happened less than a minute after loading (literally moving 5 meters from the spawn point), the second one took around 10 minutes of playing.

[NPEB01352-00.tar.gz](https://github.com/RPCS3/rpcs3/files/15319100/NPEB01352-00.tar.gz)

**Comment by valters-tomsons** [comment]
Just tested with `mesa-tkg-git` (24.2.0_devel.189413.eebc4ec264e-1). It does not seem to crash _at that specific_ place anymore, but continuing to play, it still crashes after ~5 minutes.

[log here: RPCS3.zip](https://github.com/RPCS3/rpcs3/files/15326610/RPCS3.zip)

**Comment by ro-nin** [comment]
I also had this. Playing on Steam Deck, played the game with vulkan up until entering the Hotel. THen switched to OpenGl to resolve flickering issues, but when leaving the hotel hobby it's always get stuck at the loading screen. Reverting  to vulkan makes it past the crash. v 0.0.34-1-665bb832 alpha

**Comment by valters-tomsons** [comment]
Can confirm this is still happening with appimage `0.0.35-17686-d9071abd` and mesa `25.0.1-2`. Save game attached.

[NPEB01352-00.tar.gz](https://github.com/user-attachments/files/19406590/NPEB01352-00.tar.gz)

**Comment by digant73** [comment]
@valters-tomsons, @ro-nin is that still an issue?

**Comment by valters-tomsons** [comment]
> [@valters-tomsons](https://github.com/valters-tomsons), [@ro-nin](https://github.com/ro-nin) is that still an issue?

Yes, just reproduced with `v0.0.38-18261-3c1ebe9f` AppImage on mesa `25.2.2-4`

**Comment by digant73** [comment]
please an update on current build

**Comment by valters-tomsons** [comment]
No longer reproducible on `v0.0.40-19392-0a5b88ac`
